features/steps/givens/attributes.py

```diff
@@ -7,9 +7,7 @@
 from validation_handling import gherkin_ifc
 from . import ValidationOutcome, OutcomeSeverity
 
-register_type(file_or_model=TypeBuilder.make_enum(dict(map(lambda x: (x, x), ("file", "model")))))
-register_type(plural_or_single=TypeBuilder.make_enum(dict(map(lambda x: (x, x), ("plural", "single")))))
-
+register_type(first_or_final=TypeBuilder.make_enum({"first": 0, "final": 1 }))
```

Comment:
```suggestion
from enum import Enum, auto
class FirstOrFinal(Enum):
  FIRST = auto()
  FINAL = auto()

register_type(first_or_final=TypeBuilder.make_enum({"first": FirstOrFinal.FIRST, "final": FirstOrFinal.FINAL }))
```

What do you think? I didn't test this, but I find 0 and 1 too easy to mix up.

Reply:
@Ghesselink If we leave as-is without the enum please change "final" to be -1 for consistency with python list indexing.

Reply:
I thought it would be the easiest and best readable to go along with the suggestiion of @aothms 
See commit [first_and_final adjustmenet, remove filtering by](https://github.com/buildingSMART/ifc-gherkin-rules/pull/179/commits/80533ec882a87f0d85d314ef4ce5aa6ba6bd3dd5) 

Also here the filtering_by is removed as it's not used anymore.